bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/resources/HeosStringPropertyChangeListener.java

```diff
@@ -33,14 +41,13 @@ public void removePropertyChangeListener(PropertyChangeListener listener) {
         pcs.removePropertyChangeListener(listener);
     }
 
-    public String getValue() {
+    public @Nullable String getValue() {
         return value;
     }
 
     public void setValue(String newValue) {
-        String oldValue = this.value;
         value = newValue;
-        pcs.firePropertyChange("value", oldValue, newValue);
-        value = null;
+        logger.debug("Firing property change: {} {}", newValue, Thread.currentThread());
+        pcs.firePropertyChange("value", null, newValue);
```

Comment:
This kept me busy for a long time the old code was also effectively only passing `null`. It was caused by line 44, after firing it would always reset the value (thus the future `oldValue`) to `null`